vxvas2nets/tests/test_vas2nets_sms.py

```diff
@@ -283,20 +377,67 @@ def handler(req):
             content='hi')
 
         [nack] = yield self.tx_helper.wait_for_dispatched_events(1)
-        self.assertEqual(nack['event_type'], 'nack')
-        self.assertEqual(nack['user_message_id'], msg['message_id'])
-        self.assertEqual(nack['sent_message_id'], msg['message_id'])
-        self.assertEqual(nack['nack_reason'], 'Unknown: foo')
+        self.assert_contains_items(nack, {
+            'event_type': 'nack',
+            'user_message_id': msg['message_id'],
+            'sent_message_id': msg['message_id'],
+            'nack_reason': 'Unknown request failure: foo',
+        })
+
+        [status] = self.tx_helper.get_dispatched_statuses()
+
+        self.assert_contains_items(status, {
+            'status': 'down',
+            'component': 'outbound',
+            'type': 'request_fail_unknown',
+            'message': 'Unknown request failure: foo',
+        })
 
     @inlineCallbacks
     def test_outbound_missing_fields(self):
+        yield self.mk_transport()
+
         msg = yield self.tx_helper.make_dispatch_outbound(
             from_addr='456',
             to_addr='+123',
             content=None)
 
         [nack] = yield self.tx_helper.wait_for_dispatched_events(1)
-        self.assertEqual(nack['event_type'], 'nack')
-        self.assertEqual(nack['user_message_id'], msg['message_id'])
-        self.assertEqual(nack['sent_message_id'], msg['message_id'])
-        self.assertEqual(nack['nack_reason'], 'Missing fields: content')
+        self.assert_contains_items(nack, {
+            'event_type': 'nack',
+            'user_message_id': msg['message_id'],
+            'sent_message_id': msg['message_id'],
+            'nack_reason': 'Missing fields: content',
+        })
+
+    @inlineCallbacks
+    def test_outbound_timeout(self):
+        self.remote_request_handler = lambda _: NOT_DONE_YET
+        yield self.mk_transport(outbound_request_timeout=3)
+
+        msg = self.tx_helper.make_outbound(
+            from_addr='456',
+            to_addr='+123',
+            content='hi')
+
+        self.patch_reactor_call_later()
+        d = self.tx_helper.dispatch_outbound(msg)
+
+        self.clock.advance(2)
+        self.assertEqual(self.tx_helper.get_dispatched_statuses(), [])
+
+        self.clock.advance(2)
+        yield d
+
+        [status] = self.tx_helper.get_dispatched_statuses()
+
+        self.assert_contains_items(status, {
+            'status': 'down',
+            'component': 'outbound',
+            'type': 'request_timeout',
+            'message': 'Request timeout',
+        })
```

Comment:
@hodgestar @rudigiesler I'm a bit stumped on a twisted thing here. We need to patch `reactor.callLater`, since treq uses that for request timeouts, but that seems to cause the deferred returned by `tx_helper.dispatch_outbound()` to never callback. I'm guessing patching `reactor.callLater` is breaking something somewhere in the transport, but really not sure. Any ideas?